### PR TITLE
Create files for connecting to MDSplus

### PR DESCRIPTION
**Pull request by kuchtact**
Create a file for making connection object to remote MDSplus trees/shots and a config file to be read before connecting.

### CODE REVIEW

**Comment by kuchtact** [comment]
pre-commit.ci autofix

**Comment by namurphy** [comment]
pre-commit.ci autofix

File: src/wipplpy/modules/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"""
+The `~wipplypy.modules` subpackage contains commonly used functions and classes 
+for loading and modifying data.
+"""
+__all__ = []
```
Comment (namurphy) [inline src/wipplpy/modules/__init__.py]:
```suggestion
__all__ = ["shot_loader"]
```
The `__all__` should contain all of the objects that we want to show up when we do `dir(...)`.

**Comment by kuchtact** [comment]
pre-commit.ci autofix

File: src/wipplpy/modules/shot_loader.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        Server ip address.
+    reconnect : bool, default=False
+        Whether to force a reconnection to the server.
+
+    Returns
+    -------
+    mds.Connection
+        Connection to the server without a tree connection.
+    """
+    global _mds_connection
```
Comment (namurphy) [inline src/wipplpy/modules/shot_loader.py]:
```suggestion
    global _mds_connection  # noqa: PLW0603
```
The `# noqa` comment stands for "no quality assurance" and the `PLW0604` is the specific [ruff rule](https://docs.astral.sh/ruff/rules/#pylint-pl) to ignore for this specific line.

**Pull request merged**